src/lib/transition_frontier/frontier_base/breadcrumb.ml

```diff
@@ -11,40 +11,61 @@ module T = struct
   type t =
     { validated_transition: External_transition.Validated.t
     ; staged_ledger: Staged_ledger.t sexp_opaque
-    ; just_emitted_a_proof: bool }
+    ; just_emitted_a_proof: bool
+    ; transition_receipt_time: Time.t }
```

Comment:
There are many points in the code where we are just setting this to `Time.now ()` because we did not receive this from the network. A more useful type here would be something like `type source = Genesis | Persistence | Bootstrap of Time.t | Catchup of Time.t | NetworkBroadcast of Time.t`. In this context, `Bootstrap` would represent that this transition was downloaded (not gossiped) from an RPC during the bootstrap phase, and the same goes for `Catchup` (except it was downloaded from `ledger_catchup.ml`).

If it's too difficult to get that granular for the scope of this PR, at minimum, a `Time.t option` would be a nicer way to represent this. I don't want the receipt times on these to be misleading when inspecting or debugging.

Reply:
I'll change it to an option. I believe the plan is to scuttle telemetry in mainnet, so it's not worth investing in complexity here.